scan-bar.go

```diff
@@ -43,6 +43,9 @@ func fixateScanBar(text string, width int) string {
 // Progress bar function report objects being scaned.
 type scanBarFunc func(string)
 
+func discardScanBarFactory(s string) {
```

Comment:
Yes it would be misleading to call it factory.  you can use a local 'func()' closure rather than declaring it here.